wire-ios-data-model/E2EIdentity/E2EIService.swift

```diff
@@ -0,0 +1,111 @@
+    func getNewAccountRequest(previousNonce: String) async throws -> Data
+    func setAccountResponse(accountData: Data) async throws
+
+}
+
+public final class E2EIService: E2EIServiceInterface {
+
+    // MARK: - Properties
+
+    private weak var context: NSManagedObjectContext?
+    public var wireE2eIdentity: WireE2eIdentity?
+    private let coreCrypto: SafeCoreCryptoProtocol
+
+    // MARK: - Life cycle
+
+    public init(context: NSManagedObjectContext,
+                coreCrypto: SafeCoreCryptoProtocol) {
+        self.context = context
+        self.coreCrypto = coreCrypto
+    }
+
+    // MARK: - Setup enrollment
+
+    public func setupNewEnrollment() {
+        guard
+            let context = context,
+            let selfUserHandle = ZMUser.selfUser(in: context).handle,
+            let selfUserName = ZMUser.selfUser(in: context).name,
+            let selfClient = ZMUser.selfUser(in: context).selfClient(),
+            let clientId = MLSClientID(userClient: selfClient)
+        else {
+            return
+        }
+        // TODO: we should use the new CoreCrypto version: `e2eiNewRotateEnrollment` and `e2eiNewActivationEnrollment`
+        wireE2eIdentity = try? coreCrypto.perform { try $0.e2eiNewEnrollment(clientId: clientId.rawValue,
+                                                                             displayName: selfUserName,
+                                                                             handle: selfUserHandle,
+                                                                             expiryDays: UInt32(90),
+                                                                             ciphersuite: defaultCipherSuite.rawValue)
+
+        }
+
+    }
+
+    public func isE2EIEnabled() -> Bool {
+        // TODO: we should use the new CoreCrypto version
+        // return coreCrypto.e2eiIsEnabled(defaultCipherSuite.rawValue)
+        return true
+    }
+
+    // MARK: - WireE2EIdentity methods
+
+    public func directoryResponse(directoryData: Data) throws -> WireCoreCrypto.AcmeDirectory {
+        let buffer = [UInt8](directoryData)
+        guard let wireE2eIdentity = wireE2eIdentity else {
+            WireLogger.e2ei.warn("wireE2eIdentity is missing")
+
+            throw Failure.failedToEncodeDirectoryResponse
+        }
+
+        return try wireE2eIdentity.directoryResponse(directory: buffer)
+    }
+
+    public func getNewAccountRequest(previousNonce: String) async throws -> Data {
+        guard let accountRequest = try? wireE2eIdentity?.newAccountRequest(previousNonce: previousNonce) else {
+            WireLogger.e2ei.warn("Failed to get new account request")
+
+            throw Failure.failedToGetAccountRequest
+        }
+
+        return accountRequest.data
+    }
+
+    public func setAccountResponse(accountData: Data) async throws {
+        let buffer = [UInt8](accountData)
+        try? wireE2eIdentity?.newAccountResponse(account: buffer)
+    }
```

Comment:
this method isn't throwing

Reply:
... but it should! Let the caller decide if the error should be ignored or not.
```suggestion
    public func setAccountResponse(accountData: Data) throws {
        let buffer = [UInt8](accountData)
        try wireE2eIdentity?.newAccountResponse(account: buffer)
    }
```